flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1413,4 +1415,46 @@ protected TableImpl createTable(QueryOperation tableOperation) {
 			operationTreeBuilder,
 			functionCatalog.asLookup(parser::parseIdentifier));
 	}
+
+	/**
+	 * Iterator for insert operation result.
+	 */
+	private static final class InsertResultIterator implements CloseableIterator<Row> {
```

Comment:
`TableEnvironmentImpl` is already quite large. Move it to a separate class in default scope.